chain/client-primitives/src/types.rs

```diff
@@ -63,7 +63,8 @@ pub struct DownloadStatus {
     pub state_requests_count: u64,
     pub last_target: Option<AccountOrPeerIdOrHash>,
     #[serde(skip_serializing, skip_deserializing)]
-    pub response: Arc<Mutex<Option<Result<(u16, Vec<u8>), String>>>>,
+    // Use type `String` as an error to avoid a dependency on the `rust-s3` crate.
+    pub response: Arc<Mutex<Option<Result<Vec<u8>, String>>>>,
```

Comment:
why is u16 gone?

Reply:
That `u16` was HTTP response code. Removed, because it's an implementation detail. This PR abstracts that away and provides a single interface for local filesystem and HTTP. Local filesystem operations don't have response codes.